pennylane/io/qualtran_io.py

```diff
@@ -0,0 +1,260 @@
+
+    Returns:
+        dict: A dictionary that has all the named and un-named registers with default wire
+        ordering.
+
+    Raises:
+        TypeError: bloq must be an instance of :code:`~.Bloq`.
+
+    **Example**
+
+    Given a qualtran bloq:
+
+    >>> from qualtran.bloqs.phase_estimation import RectangularWindowState, TextbookQPE
+    >>> textbook_qpe_small = TextbookQPE(ZPowGate(exponent=2 * 0.234), RectangularWindowState(3))
+    >>> qml.get_bloq_registers_info(textbook_qpe_small)
+    {'qpe_reg': Wires([0, 1, 2]), 'q': Wires([3])}
+    """
+    if not isinstance(bloq, qt.Bloq):
+        raise TypeError(f"bloq must be an instance of {qt.Bloq}.")
+
+    wire_register_dict = {reg.name: reg.bitsize for reg in bloq.signature.rights()}
+
+    return qml.registers(wire_register_dict)
+
+
+def _get_named_registers(registers):
+    """Returns a `qml.registers` object associated with the named registers in the bloq"""
+
+    temp_register_dict = {reg.name: reg.total_bits() for reg in registers}
+
+    return qml.registers(temp_register_dict)
+
+
+class FromBloq(Operation):
+    r"""
+    An adapter for using `Qualtran bloqs <https://qualtran.readthedocs.io/en/latest/bloqs/index.html#bloqs-library>`_
+    as a PennyLane :class:`~.Operation`.
+
+    Args:
+        bloq (qualtran.Bloq): the bloq to wrap
+        wires (WiresLike): the wires to act on
+
+    Raises:
+        TypeError: bloq must be an instance of :code:`~.Bloq`.
+
+    **Example**
+
+    Given a qualtran bloq:
+
+    >>> from qualtran.bloqs.basic_gates import CNOT
+
+    >>> qualtran_cnot = qml.FromBloq(CNOT(), wires=[0, 1])
+    >>> qualtran_cnot.matrix()
+    array([[1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j],
+       [0.+0.j, 1.+0.j, 0.+0.j, 0.+0.j],
+       [0.+0.j, 0.+0.j, 0.+0.j, 1.+0.j],
+       [0.+0.j, 0.+0.j, 1.+0.j, 0.+0.j]])
+
+    A simple example showcasing how to use `qml.FromBloq` inside a device:
+
+    .. code-block::
+
+        from qualtran.bloqs.phase_estimation import RectangularWindowState, TextbookQPE
+        from qualtran.bloqs.chemistry.trotter.ising import IsingXUnitary, IsingZZUnitary
+        from qualtran.bloqs.chemistry.trotter.trotterized_unitary import TrotterizedUnitary
+
+        nsites = 5
+        j_zz = 2
+        gamma_x = 0.1
+        dt = 0.01
+        indices = (0, 1, 0)
+        coeffs = (0.5 * gamma_x, j_zz, 0.5 * gamma_x)
+        zz_bloq = IsingZZUnitary(nsites=nsites, angle=2 * dt * j_zz)
+        x_bloq = IsingXUnitary(nsites=nsites, angle=0.5 * 2 * dt * gamma_x)
+        trott_unitary = TrotterizedUnitary(
+            bloqs=(x_bloq, zz_bloq), indices=indices, coeffs=coeffs, timestep=dt
+        )
+        textbook_qpe = TextbookQPE(trott_unitary, RectangularWindowState(3))
+
+        dev = qml.device("default.qubit")
+        @qml.qnode(dev)
+        def circuit():
+            qml.FromBloq(textbook_qpe, wires=range(textbook_qpe.signature.n_qubits()))
+            return qml.state()
```

Comment:
Output?

Reply:
This one's output is quite large as well.

Reply:
Same as before.